src/utils/weights.rs

```diff
@@ -63,13 +63,33 @@ impl WeightInit {
         Value::from(weight)
     }
 
-    fn he_normal(&self, fan_num: usize) -> Value {
-        let stdev = 1.0 / (fan_num as f64).sqrt();
+    fn he_normal(&self, fan_in: usize) -> Value {
+        let stdev = (2.0 / (fan_in as f64)).sqrt();
         let normal = Normal::new(0.0, stdev).unwrap();
 
         let mut rng = rand::thread_rng();
 
         let weight = normal.sample(&mut rng);
         Value::from(weight)
     }
+
+    fn lecun_uniform(&self, fan_in: usize) -> Value {
```

Comment:
How come you removed the implementation of LeCun weight initialization? It's no longer in the WeightInit enum and hence not being matched for in the `sample` method